modules/util/portal-tools-service-builder/src/main/resources/com/liferay/portal/tools/service/builder/dependencies/service_base_impl.ftl

```diff
@@ -879,9 +885,48 @@ import org.osgi.service.component.annotations.Reference;
 						${entity.varName}Persistence.closeSession(session);
 					}
 				}
+
+				@Override
+				@Transactional(readOnly = true)
+				public InputStream open${entityColumn.methodName}InputStream(long ${entity.PKVarName}) {
+					try {
+						${entity.name}${entityColumn.methodName}BlobModel
+							${entity.name}${entityColumn.methodName}BlobModel = get${entityColumn.methodName}BlobModel(
+								${entity.PKVarName});
+
+						Blob blob = ${entity.name}${entityColumn.methodName}BlobModel.get${entityColumn.methodName}Blob();
+
+						InputStream inputStream = blob.getBinaryStream();
```

Comment:
See https://github.com/brianchandotcom/liferay-portal/pull/81188
We need to add a null check here for the blob reference. On oracle, when the blob is empty, oracle driver will make it a null rather than empty blob. So on seeing null blob, we should return an empty inputstream.